### PR TITLE
test-execute without capsh

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by martinpitt** [comment]
Strictly speaking this isn't a new build dependency, as test-execute is already skipped when run as non-root (i. e. during a normal make/make check). I must say I'd rather have it fail with a better error message than silently skipping the tests -- otherwise it's very easy to miss that, and the test coverage is incomplete.

File: src/test/test-execute.c [hunk 1]
```diff
@@ -183,6 +195,13 @@ static void test_exec_protectkernelmodules(Manager *m) {
                 return;
         }
 
+        r = find_binary("capsh", NULL);
+        if (r < 0) {
+                log_error_errno(r, "Skipping %s, could not find capsh binary: %m", __func__);
+                return;
+        }
+
+
```
Comment (poettering) [inline src/test/test-execute.c:204]: not that it matters, but: spurious double newline...

**Comment by poettering** [comment]
anyway, this looks good to me. Merging this now. I don't think it should be necessary to install the caps tools just to make our make check pass. (speaking as a developer, not as a packager/tester in this case)

That said, I also think @martinpitt has a point, my proposal would be to maybe add a configure check that warns loudly if capsh is missing at build time, and clarifies that some tests are skipped due to this?

**Comment by keszybz** [comment]
Oh, I reworked the PR in the mean time. I don't care much either way.

Instead of trying to add configure switches with warnings for each and every contingency, I'd rather see us convert to TAP and use their SKIP directive, which outputs an explanation. It'd probably be also possible to provide a summary of skipped tests (15 skipped: root required, capsh missing, foobar too old). I think that'd be much more scalable and nicer for users.

**Pull request merged**